federation/src/watcher.c

```diff
@@ -0,0 +1,179 @@
+#include "watcher.h"
+
+spiffebundle_Watcher *spiffebundle_Watcher_New()
+{
+    spiffebundle_Watcher *watcher
+        = (spiffebundle_Watcher *) calloc(1, sizeof(*watcher));
+    sh_new_strdup(watcher->endpoints);
+    return watcher;
+}
+
+void spiffebundle_Watcher_Free(spiffebundle_Watcher *watcher)
+{
+    if(watcher) {
+        for(size_t i = 0, length = shlenu(watcher->endpoints); i < length;
```

Comment:
```suggestion
        for(size_t i = 0, size = shlenu(watcher->endpoints); i < size;
```
Just for consistency